src/modm/platform/eth/stm32/module.lb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016-2018, Niklas Hauser
+# Copyright (c) 2017, Fabian Greif
+# Copyright (c) 2020, Mike Wolfram
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":platform:eth"
+    module.description = "Ethernet of STM32F4/F7"
+
+def prepare(module, options):
+    device = options[":target"]
+
+    # FIXME the driver is for L4 only
+    if device.identifier["family"] not in ["f7"]:
```

Comment:
Do all F7's have Ethernet? You could additionally check the device files with device.has_driver("eth:stm32") or similiar.

Reply:
You can run the test locally with `(cd test/all; python3 run_all.py stm32f7)`. That's faster than waiting for the CI, since it remembers the failed ones and retries only those on the next call.

Reply:
Added check for driver.